nrf_modem/include/nrf_modem_full_dfu.h

```diff
@@ -0,0 +1,221 @@
+
+#include <stdint.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(__GNUC__) || (__GNUC__ == 0)
+typedef int32_t ssize_t;
+#else
+#include <sys/types.h>
+#ifdef __SES_ARM
+typedef int32_t ssize_t;
+#endif
+#endif
+
+/**@addtogroup nrf_full_dfu_api_utils
+ *@{
+ */
+
+/**@brief Digest buffer length. */
+#define NRF_MODEM_FULL_DFU_DIGEST_LEN 32
+/**@brief UUID buffer length. */
+#define NRF_MODEM_FULL_DFU_UUID_LEN   36
+
+/**@brief Required shared memory length */
+#define NRF_MODEM_FULL_DFU_SHM_LEN 0x201c
+
+/**@brief Storage for 256-bit digest/hash replies.
+ * Endianness not converted.
+ */
+struct nrf_modem_full_dfu_digest {
+	uint8_t data[NRF_MODEM_FULL_DFU_DIGEST_LEN];
+};
+
+/**@brief Modem UUID response. */
+struct nrf_modem_full_dfu_uuid {
+	uint8_t data[NRF_MODEM_FULL_DFU_UUID_LEN];
+};
+
+/**@brief Set modem in full DFU mode.
+ *
+ * Reset the modem and set it into a state ready for full DFU operation.
+ * Before initialization, the modem library must be shut down and
+ * reinitialized with FULL_DFU_MODE.
+ *
+ * Call once before DFU operations. If the modem goes to a error state,
+ * this can be called again to re-initialize.
+ * The root key digest response of the modem is put in the digest_buffer
+ * structure.
+ *
+ * If success, the modem will be in a state to wait for a bootloader.
+ * Follow up with a call to nrf_modem_full_dfu_bl_write().
+ *
+ * @param[out]  digest_buffer       Pointer to the buffer to store digest hash.
+ *                                  If NULL, digest skipped.
+ *
+ * On failure, may set errno to one of following values:
+ *
+ * - \c NRF_EFAULT When modem responded with error codes.
+ * - \c NRF_ETIMEDOUT When modem did not respond.
+ * - \c NRF_EIO When incorrect response received after modem reset.
+ *
+ * @return 0 if the procedure succeeds, else, -1 and errno set in case of error.
+ */
+int nrf_modem_full_dfu_init(struct nrf_modem_full_dfu_digest *digest_buffer);
+
+/**@brief Write a booloader chunk to the modem.
+ *
+ * Call after nrf_modem_full_dfu_init() to upload modem bootloader segments.
+ * Bootloader segments are appended together, so the full bootloader
+ * may be uploaded with one or many call to this function.
+ * Once the full bootloader is uploaded, call nrf_modem_full_dfu_apply().
+ *
+ * Firmware segments can be uploaded after successful bootloader upload.
+ *
+ * On failure, may set errno to one of following values:
+ *
+ * - \c NRF_EINVAL When src pointer is NULL or user tries to upload too big
+ *                 bootloader.
+ * - \c NRF_EOPNOTSUPP If the modem is not in a state to receive.
+ * - \c NRF_EFAULT When modem responded with error codes.
+ * - \c NRF_EPERM When modem did not accept the flash programming request.
+ * - \c NRF_ENOEXEC When flash programming failed.
+ * - \c NRF_ETIMEDOUT When modem did not respond.
+ * - \c NRF_EIO When incorrect response received from modem.
+ *
+ * @param[in]   addr   Address where the data to be written.
```

Comment:
There's no such parameter in this function.

Reply:
Good catch, that parameter have been removed from bootloader write.

@lemrey Where should I send a patch, or do you want to do a quick fixup in your branch? This is just one line to be removed from Doxygen block.

Reply:
@SeppoTakalo I can take it, no problem